src/main/java/de/tum/in/www1/artemis/web/rest/GenericSubmissionResource.java

```diff
@@ -0,0 +1,62 @@
+package de.tum.in.www1.artemis.web.rest;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.Exercise;
+import de.tum.in.www1.artemis.domain.Submission;
+import de.tum.in.www1.artemis.service.*;
+import de.tum.in.www1.artemis.web.rest.util.HeaderUtil;
+
+/**
+ * Generic Resource for Text, Modeling and File Upload Submission REST Controllers
+ */
+public abstract class GenericSubmissionResource<T extends Submission, E extends Exercise> {
```

Comment:
There still is code duplication in the submission resources, e.g. `get...SubmissionWithoutAssessment` and `getAll...Submissions`. I would move them to this generic resource trying to pull out as much generic code as possible. You can then still call the generic method and do some specialized operations afterwards like hiding specific details or having some additional authorization checks

Reply:
Thank you for the ideas. Indeed I found a way to generalize the logic even more